spiketoolkit/validation/quality_metric_classes/utils/curationsortingextractor.py

```diff
@@ -236,9 +236,36 @@ def split_unit(self, unit_id, indices):
             self._roots.append(new_root_2)
 
             for feature_name in self.get_unit_spike_feature_names(unit_id):
-                full_features = np.array(self.get_unit_spike_features(unit_id, feature_name))
-                self.set_unit_spike_features(new_root_1_id, feature_name, full_features[indices_1])
-                self.set_unit_spike_features(new_root_2_id, feature_name, full_features[indices_2])
+                if feature_name.endswith('_idxs'):
+                    continue
+                full_features = self.get_unit_spike_features(unit_id, feature_name)
+                if isinstance(full_features, list):
+                    full_features = np.array(full_features)    
+                if not feature_name + '_idxs' in self.get_unit_spike_feature_names(unit_id):
+                    self.set_unit_spike_features(new_root_1_id, feature_name, full_features[indices_1])
+                    self.set_unit_spike_features(new_root_2_id, feature_name, full_features[indices_2])
+                else:
+                    full_features_idxs = np.array(self.get_unit_spike_features(unit_id, feature_name+'_idxs'))
+                    indices_1_idxs = [n for n, i in enumerate(full_features_idxs) if i in indices]
```

Comment:
these should be parsed as np.array otherwise using them as idxs will fail in line 263:

`indices_1_idxs = [n for n, i in enumerate(full_features_idxs) if i in indices]`
`indices_2_idxs = [n for n, i in enumerate(full_features_idxs) if i not in indices]`